kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,190 @@
+import os
+import re
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            '"%s" is similar to "%s". Try keeping them the same to reduce work for translators!',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    _dict_of_strings = {}
+    threshold = 0.8
+    error_messages = {}
+
+    project_root = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), "..", "tcms")
+    )
+
+    # NOTE: this works against tcms/ directory and will not take into account
+    # if we want to examine only a sub-dir or a few files
+    # all files found by os.walk
+    all_template_files = set()
+
+    def open(self):
+        for rootpath, _dirs, files in os.walk(self.project_root, topdown=False):
+
+            for file_name in files:
+                if file_name.endswith(".html"):
+                    self.all_template_files.add(
+                        os.path.join(self.project_root, rootpath, file_name)
+                    )
+        for rootpath, _dirs, files in os.walk(
+            os.path.join(self.project_root, "templates", "email"), topdown=False
+        ):
+
+            for file_name in files:
+                if file_name.endswith(".txt"):
+                    self.all_template_files.add(
+                        os.path.join(self.project_root, rootpath, file_name)
+                    )
```

Comment:
Is there a reason for the txt file scanning to be inside a second for loop instead of just moving the if block above under the original loop ?

Reply:
Still unresolved, you've changed the places of the 2 for loops but didn't address the original comment.

Reply:
Actually, I wanted to only detect .txt files in `templates/email` folder because other txt files are not template files for ex - there are required dependency .txt files in `requirements` folder and there is .txt file in `tests/bugzilla` folder too. Although there won't be any such tag like trans and blocktrans in them but I think we shouldn't unnecessarily scan them if they aren't template files. And after making the change i.e changing placement of the loop, we're detecting that if the current root is `templates/email` then only we'll add txt files to the `all_template_files` set. Previously, there was another loop other than previous loop to only parse through `template/email` folder.

Reply:
FTR other .txt files will not have the trans or blocktrans tags inside of them so they shouldn't affect the results.